react/react1/week2/todo-app/src/components/TodoRow.js

```diff
@@ -0,0 +1,30 @@
+import Button from './Button';
+import { useState } from 'react';
+import TodoItem from './TodoItem';
+import styled from 'styled-components';
+import TodoCheckbox from './TodoCheckbox';
+
+const TodoRowBox = styled.div`
+  display: flex;
+  align-items: center;
+`;
+
+const TodoRow = (props) => {
+  const { item, index, updateTodoList } = props;
+
+  const [checked, setChecked] = useState(false);
```

Comment:
what you are doing here works great, but you are not really changing the todo list data. The better way is to have this in your todo item and change that once the user click on the checkbox, so the todo item will look something like this: 
`{
    id: 1,
    description: 'Get out of bed',
    completed: true
  },`
then you can have a function so it would update that specific todo item. That's more likely how it would be in the real life scenario.

Reply:
Okay, let's imagine, those todos which you have in your data folder, is coming from an API, and it's not a hardcoded data. with the current setup, once they click on the checkbox to mark it as completed, the state `checked` will be updated and mark it as checked. what happens when you refresh the page ? the state gets cleared, and therefore everything will be fetched from the API again, and that specific item is not checked again. so it's better to include a property as `completed` in your data, and update that property from true to false or versa when they click on the check box, in the real world this will trigger an API call the update the status of the todo item, and even once you refresh your page the newly fetched data will have your `completed` task up to date.
It's not a necessary thing to do here, but just to have this in your mind for future projects.